### PR TITLE
(CDAP-7500) Fix premature closing of MR classloader

### PR DESCRIPTION
**Pull request by chtyim**
- Reverted CDAP-7392 fix to use shutdown hook for closing CDAP Services
- Create Kafka Producer from daemon thread
  - This will ensure the async ProducerSendThread created from the kafka Producer constructor
    is also a daemon thread (from Java Thread class javadoc)
- Also fixed an annoying and misleading exception from Kafka when the caller thread is interrupted.

### CODE REVIEW

**Comment by chtyim** [comment]
Build: http://builds.cask.co/browse/CDAP-RUT244-1

**Comment by chtyim** [comment]
Build passed.

**Comment by sagarkapare** [comment]
💯

File: cdap-watchdog/src/main/java/co/cask/cdap/logging/appender/kafka/SimpleKafkaProducer.java [hunk 1]
```diff
@@ -53,19 +58,44 @@ public SimpleKafkaProducer(CConfiguration cConf) {
...
     }
   }
 
   public void stop() {
     producer.close();
   }
+
+  /**
+   * Creates a {@link Producer} using the given configuration. The producer instance will be created from a
+   * daemon thread to make sure the async thread created inside Kafka is also a daemon thread.
```
Comment (anew) [inline cdap-watchdog/src/main/java/co/cask/cdap/logging/appender/kafka/SimpleKafkaProducer.java:86]: nice!

**Review by anew — APPROVED** [review]
LGTM

File: cdap-watchdog/src/main/java/co/cask/cdap/logging/appender/kafka/SimpleKafkaProducer.java [hunk 2]
```diff
@@ -53,19 +58,44 @@ public SimpleKafkaProducer(CConfiguration cConf) {
...
+    boolean threadInterrupted = Thread.interrupted();
     try {
       KeyedMessage<String, byte[]> data = new KeyedMessage<>(kafkaTopic, key, bytes);
       producer.send(data);
     } catch (Throwable t) {
       LOG.error("Exception when trying to publish log message to kafka with key {} and topic {}", key, kafkaTopic, t);
+    } finally {
+      // Reset the interrupt flag if needed
+      if (threadInterrupted) {
+        Thread.currentThread().interrupt();
```
Comment (gsps1) [inline cdap-watchdog/src/main/java/co/cask/cdap/logging/appender/kafka/SimpleKafkaProducer.java:75]: can you add a comment for why we call interrupt here ?

Comment (chtyim) [reply cdap-watchdog/src/main/java/co/cask/cdap/logging/appender/kafka/SimpleKafkaProducer.java:75]: Hum? It already has the comment "Reset the interrupt flag if needed". It is based on the `threadInterrupted` local variable that get set right above.

**Pull request merged**